### PR TITLE
allow signed url integration tests to work with MANTA_TLS_INSECURE

### PR DESCRIPTION
**Pull request by cburroughs**
ref #525

### CODE REVIEW

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java [hunk 1]
```diff
@@ -23,12 +23,21 @@
 import java.net.HttpURLConnection;
 import java.net.URI;
 import java.nio.charset.Charset;
+import java.security.SecureRandom;
+import java.security.SecureRandom;
```
Comment (nairashwin952013) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java]: Redundant Import

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java [hunk 2]
```diff
@@ -65,6 +74,51 @@ public void afterClass() throws IOException {
...
+                    @Override
+                    public void checkClientTrusted(X509Certificate[] certs, String authType) {
+                    }
+                    @Override
+                    public void checkServerTrusted(X509Certificate[] certs, String authType) {
+                    }
+                }
+            };
+
+             HostnameVerifier hostnameVerifier = new HostnameVerifier() {
```
Comment (nairashwin952013) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java]: This code block could be replaced with a lambda

```
HostnameVerifier hostnameVerifier = new HostnameVerifier() {
                @Override
                public boolean verify(String hostname, SSLSession session) {
                    return true;
                }
            };
```
to 

`HostnameVerifier hostnameVerifier = (hostname, session) -> true;`

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java [hunk 3]
```diff
@@ -65,6 +74,51 @@ public void afterClass() throws IOException {
...
+                @Override
+                public boolean verify(String hostname, SSLSession session) {
+                    return true;
+                }
+            };
+
+            try {
+                SSLContext sc = SSLContext.getInstance("SSL");
```
Comment (nairashwin952013) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java]: apologies for the annoying nit but could we *refactor* `sc` & `sConn` to more readable variables.

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java [hunk 4]
```diff
@@ -65,6 +74,51 @@ public void afterClass() throws IOException {
...
     }
 
+
+    // When MantaClient TLS security is disabled (such as when testing against a
+    // lab instance without a self signed cert), we will also need to refrain
+    // from checking the certs on the resulting signed urls.
+    private HttpURLConnection openHttpURLConnection(URI uri, MantaClient client) throws IOException {
+        boolean tlsInsecure = client.getContext().tlsInsecure();
+        HttpURLConnection connection = (HttpsURLConnection)uri.toURL().openConnection();
+        if (tlsInsecure && connection instanceof HttpsURLConnection) {
```
Comment (nairashwin952013) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java]: Doubt: Could we replace `instanceof` condition with a `null` check ?

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java [hunk 5]
```diff
@@ -65,6 +74,51 @@ public void afterClass() throws IOException {
         IntegrationTestConfigContext.cleanupTestDirectory(mantaClient, testPathPrefix);
     }
 
+
+    // When MantaClient TLS security is disabled (such as when testing against a
```
Comment (nairashwin952013) [inline java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java]: Comment could be replaced as a `javadoc`.

File: java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java (hunk 4 above)
Comment (cburroughs) [reply java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSigningIT.java]: The `instanceof` is for checking to make sure it is an `HttpsURLConnection` (subclass of `HttpURLConnection`).  I think almost all manta standups -- even lab ones -- use https, but if someone had an http only dev setup I didn't want to break that.

**Pull request merged**